cdap-docs/build.sh

```diff
@@ -460,18 +497,128 @@ function copy_docs_inner_level() {
   fi
-  echo "Adding .htaccess file (404 file)"
-  rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/${PROJECT_VERSION}/.${HTACCESS} "<version>" "${PROJECT_VERSION}"
   cd ${TARGET_PATH}
-  zip -qr ${ZIP_DIR_NAME}.zip ${PROJECT_VERSION}/.${HTACCESS}
+  doc_change_py="${TARGET_PATH}/../tools/doc-change.py"
+
+  echo "Removing old directories and zips"
+  rm -rf base current ${PROJECT_VERSION} *.zip
+  
+  echo "Creating base"
+  mkdir base
+  cp -r ${HTML} base/en
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not create base"
+      return ${errors}   
+  fi
+
+  # Add a redirect index.html file
+  cp ${SCRIPT_PATH}/${COMMON_SOURCE}/redirect.html base/index.html
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not add redirect file"
+      return ${errors}   
+  fi
+
+  echo
+  echo "1: 'current' version, with canonical ref link ${zip_dir_name}-current"
+  cp -r base current
+  python ${doc_change_py} ${build_flag} --type=current --version=${PROJECT_VERSION} ${TARGET_PATH}/current
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not change doc set ${TARGET_PATH}/current"
+      return ${errors}   
+  fi
+  zip -qr ${zip_dir_name}.zip current/* --exclude *.DS_Store* *.buildinfo*
+  if [[ "x${zip_extras}" != "x" ]]; then
+    echo "Creating and adding a .htaccess file (404 file)"
+    rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/current/.${HTACCESS} "<version>" "current"
+    zip -qr ${zip_dir_name}.zip current/.${HTACCESS}
+  fi
+  mv ${zip_dir_name}.zip ${zip_dir_name}-current.zip
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not move ${zip_dir_name}.zip to ${zip_dir_name}-current.zip"
+      return ${errors}   
+  fi
+
+  echo
+  echo "2: 'future' version with robots meta tag ${zip_dir_name}-future"
+  cp -r base ${PROJECT_VERSION}
+  python ${doc_change_py} ${build_flag} --type=future --version=${PROJECT_VERSION} ${TARGET_PATH}/${PROJECT_VERSION}
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not change doc set ${TARGET_PATH}/${PROJECT_VERSION}"
+      return ${errors}   
+  fi
+  zip -qr ${zip_dir_name}.zip ${PROJECT_VERSION}/* --exclude *.DS_Store* *.buildinfo*
+  if [[ "x${zip_extras}" != "x" ]]; then
+    echo "Creating and adding a .htaccess file (404 file)"
+    rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/${PROJECT_VERSION}/.${HTACCESS} "<version>" "${PROJECT_VERSION}"
+    zip -qr ${zip_dir_name}.zip ${PROJECT_VERSION}/.${HTACCESS}
+  fi
+  mv ${zip_dir_name}.zip ${zip_dir_name}-future.zip
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not move ${zip_dir_name}.zip to ${zip_dir_name}-future.zip"
+      return ${errors}   
+  fi
+  rm -rf future
+  mv ${PROJECT_VERSION} future
+  errors=$?
+  if [[ ${errors} -ne 0 ]]; then
+      echo "Could not move ${PROJECT_VERSION} to future"
+      return ${errors}   
+  fi
```

Comment:
Because then you (I) can look at the directory results and check that the contents are correct, either locally on on the Bamboo build. True, you should really download the zips and unpack them (which I do as a final check) but while I am getting this going, it allows for an intermediate check that things are ok.